packages/tasit-action/src/TasitAction.js

```diff
@@ -107,12 +117,51 @@ class TransactionSubscription extends Subscription {
     this.#provider = provider;
   }
 
-  #addConfirmationListener = async listener => {
+  on = (eventName, listener) => {
+    this.#addListener(eventName, listener, false);
+  };
+
+  once = (eventName, listener) => {
+    this.#addListener(eventName, listener, true);
+  };
+
+  getEventsTimeout = () => {
+    return this.#timeout;
+  };
+
+  setEventsTimeout = timeout => {
+    this.#timeout = timeout;
+  };
+
+  #addListener = (eventName, listener, once) => {
+    const triggers = ["confirmation", "error"];
+
+    if (!triggers.includes(eventName))
+      throw new Error(`Invalid listener trigger, use: [${triggers}]`);
+
+    if (eventName === "error" && once)
+      throw new Error(`Use on() function to subscribe an error event.`);
+
+    if (!listener || typeof listener !== "function")
+      throw new Error(`Cannot listen without a function`);
+
+    if (eventName === "error") {
+      this._addErrorListener(listener);
+    } else if (eventName === "confirmation") {
+      this.#addConfirmationListener(listener, once);
+    }
```

Comment:
Sounds good to me!

Reply:
And maybe handle what's else case first (this would require changing it to an if with inverted logic) to follow the practice of handling error cases early and often before moving onto the main code